### PR TITLE
partners:MongoDB New Checkpointer

### PR DESCRIPTION
**Pull request by caseyclements**
Implements two new graph checkpointers for MongoDB. One synchronous and one asynchronous.
This closely follows the other implementations in checkpoint, checkpoint-postgres, checkpoint-sqlite but with NoSQL idioms.

### CODE REVIEW

File: libs/checkpoint-mongodb/langgraph/checkpoint/mongodb/__init__.py [hunk 1]
```diff
@@ -0,0 +1,311 @@
+"""
+TODO: Update Docs
```
Comment (blink1073) [inline libs/checkpoint-mongodb/langgraph/checkpoint/mongodb/__init__.py:2]: Is this TODO for this PR?

File: libs/checkpoint-mongodb/langgraph/checkpoint/mongodb/aio.py [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+    Dict,
+    Optional,
+    Sequence,
+    Tuple,
+    Union,
+)
+
+from langchain_core.runnables import RunnableConfig
+from pymongo import AsyncMongoClient, UpdateOne
+from pymongo.asynchronous.database import AsyncDatabase
```
Comment (blink1073) [inline libs/checkpoint-mongodb/langgraph/checkpoint/mongodb/aio.py:14]: We should be using Motor until we take Async PyMongo out of beta

File: libs/checkpoint-mongodb/tests/test_async.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+)
+from langgraph.checkpoint.mongodb.aio import AsyncMongoDBSaver
+
+# Setup:
+# docker pull mongodb/mongodb-atlas-local:latest
+# docker run --name mongodb -d -p 27017:27017 mongodb/mongodb-community-serve
+MONGODB_URI = "mongodb://localhost:27017"
+DB_NAME = "langgraph-test"
+
+# TODO: Test
```
Comment (blink1073) [inline libs/checkpoint-mongodb/tests/test_async.py:21]: TODO for this PR?

File: libs/checkpoint-mongodb/langgraph/checkpoint/mongodb/__init__.py (hunk 1 above)
Comment (caseyclements) [reply libs/checkpoint-mongodb/langgraph/checkpoint/mongodb/__init__.py:2]: The docstrings, probably

File: libs/checkpoint-mongodb/langgraph/checkpoint/mongodb/aio.py (hunk 2 above)
Comment (caseyclements) [reply libs/checkpoint-mongodb/langgraph/checkpoint/mongodb/aio.py]: I switched to Motor.

File: libs/checkpoint-mongodb/langgraph/checkpoint/mongodb/__init__.py (hunk 1 above)
Comment (caseyclements) [reply libs/checkpoint-mongodb/langgraph/checkpoint/mongodb/__init__.py]: Updated the docstrings.

**Review by blink1073 — APPROVED** [review]
LGTM!

**Comment by vbarda** [comment]
Hi folks @caseyclements @blink1073 -- thank you for adding the implementation! We're currently not accepting PRs for addinng checkpointers libraries into the LangGraph repository, but we're definitely interested in the official library from you. Could you please add it in one of the Mongo repositories?

To verify that the checkpointer is functioning correctly, it needs to pass tests from [this file for sync](https://github.com/langchain-ai/langgraph/blob/main/libs/langgraph/tests/test_pregel.py) and [this file for async](https://github.com/langchain-ai/langgraph/blob/main/libs/langgraph/tests/test_pregel_async.py)

We are thinking of making the standard test more accessible for folks developing checkpointer libraries.

Please let me know if you have any questions / concerns -- happy to help!

In the meantime, will close this PR -- feel free to tag me on the one you create in a different repo.

**Comment by prakul** [comment]
Hi @vbarda Given LangGraph JS already contains a mongoDB checkpointer implementation, would it make sense to maintain a standard convention? We have the requests from a few joint customer, including few who have reached out to Langchain directly as well.

**Pull request closed without merging**